File: set_diesel_toml.py
import os

# Set the current directory to the root of the workspace
# Assuming this script is located somewhere under the workspace
current_directory = os.path.abspath(os.path.join(os.path.dirname(__file__)))

# Read the TOML file
with open('src-tauri/diesel.toml', 'r') as file:
    config_content = file.readlines()

# Modify the specific line containing the migrations directory
for i, line in enumerate(config_content):
    if line.startswith('dir ='):
        # Replace the static path with the dynamic current directory path
        config_content[i] = f'dir = "{os.path.join(current_directory, "src-tauri", "migrations")}"\n'

# Optionally, print the modified content or save it back to a file
print(''.join(config_content))

with open('src-tauri/diesel.toml', 'w') as file:
    file.writelines(config_content)

File: set_diesel_toml.mjs
import fs from "fs/promises";
import path from "path";

// Set the current directory to the root of the workspace
const currentDirectory = path.resolve();

// Function to log messages to the log file
async function logMessage(message) {
  const logFilePath = path.join(currentDirectory, "log.txt");
  const timestamp = new Date().toISOString();
  await fs.appendFile(logFilePath, `[${timestamp}] ${message}\n`, "utf8");
}

async function updateToml() {
  try {
    // Read the TOML file
    const tomlFilePath = path.join(currentDirectory, "src-tauri", "diesel.toml");
    let data = await fs.readFile(tomlFilePath, "utf8");

    // Modify the specific line containing the migrations directory
    const migrationsDir = path.join(currentDirectory, "src-tauri", "migrations");
    const configLines = data.split("\n").map(line => {
      if (line.startsWith("dir =")) {
        // Ensure the path is correctly formatted for the OS
        return `dir = "${migrationsDir.replace(/\\/g, "/")}"`; // Use forward slashes for TOML
      }
      return line;
    });

    const modifiedContent = configLines.join("\n");

    // Log the modified content (optional)
    await logMessage("Modified content:\n" + modifiedContent);

    // Write the modified content back to the TOML file
    await fs.writeFile(tomlFilePath, modifiedContent);
    await logMessage("File updated successfully.");

  } catch (err) {
    await logMessage(`Error: ${err.message}`);
  }
}

updateToml();
